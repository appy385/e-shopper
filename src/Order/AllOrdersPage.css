.all-orders {
    font-size: larger;
    padding-left: 1.25rem;
}
.past-orders {
    font-size: larger;
    padding-top: 1.8rem;
    padding-left: 1.8rem;
}
.order-table-container {
    overflow-x:auto;
}
.order-table{
    width: 50%;
    margin: auto;
    border-collapse: collapse;
    border: 1px solid #ddd;
}
.order-table-header>th {
    color: white;
    background-color: #303030;
    border-top: 1px solid #ddd;
    border-bottom: 1px solid #ddd;
    padding: 1rem;
    text-align: left;
  
}
.order-details>td {
    padding:  1.25rem;
    text-align: left;
    max-width: 1.5rem;
}
.order-item-table{
    width: 50%;
    margin: auto;
    border-collapse: collapse;
    border: 1px solid #ddd;
}
.order-category{
    background-color: lightgray;
}
.order-category>td {
    padding: 1rem;
    text-align: left;
}
.order-item>th {
    background-color: cornflowerblue;
    padding: 1rem;
    text-align: left;
    color: white;
}


