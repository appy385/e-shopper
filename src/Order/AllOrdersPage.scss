.all-orders {
    font-size: larger;
    padding-left: 1.25rem;
}
.past-orders {
    font-size: larger;
    padding-top: 1.8rem;
    padding-left: 1.8rem;
}
.order-table {
    width: 50%;
    table-layout: auto;
    margin: auto;
    border-collapse: collapse;
    border: 1px solid #ddd;
    
}
.order-table-header>th {
    color: white;
    background-color: #303030;
    border-top: 1px solid #ddd;
    border-bottom: 1px solid #ddd;
    padding: 1rem;
    text-align: left;
  
}
.order-details>td{
    padding: 1rem;
    text-align: left;
}


.order-item-container {
    overflow-x:auto;
    margin-bottom: 3rem;
}

.order-item-table{
    width: 50%;
    table-layout: auto;
    margin: auto;
    border-collapse: collapse;
    border: 1px solid #ddd;
   
}
.order-category{
    background-color: lightgray;
}
.order-category>td {
    padding: 1rem;
    text-align: left;
}
.order-item>th {
    background-color: cornflowerblue;
    padding: 1rem;
    text-align: left;
    color: white;
}
.order-item-total>td {
    padding: 1rem;
    text-align: left;

}


