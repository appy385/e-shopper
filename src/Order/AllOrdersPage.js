/* eslint-disable react/prop-types */
import React from 'react';
import './AllOrdersPage.scss';
import Item from '../Item/Item';

const AllOrdersPage = ({ orders }) => (
  <div>
    <p className="all-orders">All Orders</p>
    <hr />
    <p className="past-orders">{`Past Orders(${orders.length})`}</p>
    <div className="order-table-container">
      {
          orders.map((order) => {
            const orderDate = new Date(order.date);
            return (
              <React.Fragment key={order.id}>
                <table className="order-table">
                  <thead>
                    <tr className="order-table-header">
                      <th>Order</th>
                      <th>Item</th>
                      <th>Date</th>
                      <th>Amount</th>
                    </tr>
                  </thead>
                  <tbody>
                    <tr className="order-details">
                      <td>{`order id: ${order.id}`}</td>
                      <td>{order.total}</td>
                      <td>{(`${orderDate.getDate()}-${orderDate.getMonth()}-${orderDate.getFullYear()}`)}</td>
                      <td>{order.cost}</td>
                    </tr>
                  </tbody>
                </table>
                <div className="order-item-container">
                  <table className="order-item-table">
                    <thead>
                      <tr className="order-item">
                        <th>Item Description</th>
                        <th>Unit Price</th>
                        <th>Quantity</th>
                        <th>SubTotal</th>
                      </tr>
                    </thead>
                    <tbody>
                      {
                         Object.entries(order.items).map((CategoryItems) => (
                           <React.Fragment key={CategoryItems[0]}>
                             <tr className="order-category">
                               <td>{CategoryItems[0]}</td>
                               <td> </td>
                               <td> </td>
                               <td> </td>
                             </tr>

                             { CategoryItems[1].map((item) => (
                               <Item key={item.id} product={item} />
                             ))}
                           </React.Fragment>

                         ))
                      }

                      <tr className="order-item-total">
                        <td> </td>
                        <td> </td>
                        <td>Total:</td>
                        <td>{order.cost}</td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </React.Fragment>
            );
          })
       }
    </div>
  </div>
);
export default AllOrdersPage;
