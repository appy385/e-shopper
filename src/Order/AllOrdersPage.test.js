import React from 'react';
import { render, screen } from '@testing-library/react';
import AllOrdersPage from './AllOrdersPage';

describe(AllOrdersPage.name, () => {
  let mockOrders;
  beforeEach(() => {
    mockOrders = [
      {
        id: 1,
        iat: 134526668,
        length: 4,
        cost: 40,
        items: {
          'Fruits & Vegatables': [{
            id: 1,
            src: 'https://images.pexels.com/photos/61127/pexels-photo-61127.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500',
            seller: 'Fresho',
            name: 'Banana-Robusta',
            quantity: 10,
            price: 40,
            count: 0,
            category: 'Fruits & Vegatables',
          }],

        },
      },
      {
        id: 2,
        iat: 134526668,
        length: 4,
        cost: 40,
        items: {
          'Fruits & Vegatables': [{
            id: 1,
            src: 'https://images.pexels.com/photos/61127/pexels-photo-61127.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500',
            seller: 'Fresho',
            name: 'Banana-Robusta',
            quantity: 10,
            price: 40,
            count: 0,
            category: 'Fruits & Vegatables',
          }],

        },
      },
    ];
  });
  test('should match snapshot for AllOrdersPage component', () => {
    const { container } = render(<AllOrdersPage
      orders={mockOrders}
    />);
    expect(container).toMatchSnapshot();
  });
  test('should dispay number of orders', () => {
    render(<AllOrdersPage
      orders={mockOrders}
    />);
    screen.getByText(`Past Orders(${mockOrders.length})`);
  });
});
