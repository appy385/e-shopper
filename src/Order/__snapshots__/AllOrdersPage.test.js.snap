// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllOrdersPage should match snapshot for AllOrdersPage component 1`] = `
<div>
  <div>
    <p
      class="all-orders"
    >
      All Orders
    </p>
    <hr />
    <p
      class="past-orders"
    >
      Past Orders(2)
    </p>
    <div
      class="order-table-container"
    >
      <table
        class="order-table"
      >
        <thead>
          <tr
            class="order-table-header"
          >
            <th>
              Order
            </th>
            <th>
              Item
            </th>
            <th>
              Date
            </th>
            <th>
              Amount
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="order-details"
          >
            <td>
              order id: 1
            </td>
            <td />
            <td>
              NaN-NaN-NaN
            </td>
            <td>
              40
            </td>
          </tr>
        </tbody>
      </table>
      <div
        class="order-item-container"
      >
        <table
          class="order-item-table"
        >
          <thead>
            <tr
              class="order-item"
            >
              <th>
                Item Description
              </th>
              <th>
                Unit Price
              </th>
              <th>
                Quantity
              </th>
              <th>
                SubTotal
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              class="order-category"
            >
              <td>
                Fruits & Vegatables
              </td>
              <td>
                 
              </td>
              <td>
                 
              </td>
              <td>
                 
              </td>
            </tr>
            <tr
              class="item"
            >
              <td>
                Banana-Robusta
              </td>
              <td>
                40
              </td>
              <td>
                0
              </td>
              <td
                data-testid="item-amount"
              >
                0
              </td>
            </tr>
            <tr
              class="order-item-total"
            >
              <td>
                 
              </td>
              <td>
                 
              </td>
              <td>
                Total:
              </td>
              <td>
                40
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <table
        class="order-table"
      >
        <thead>
          <tr
            class="order-table-header"
          >
            <th>
              Order
            </th>
            <th>
              Item
            </th>
            <th>
              Date
            </th>
            <th>
              Amount
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="order-details"
          >
            <td>
              order id: 2
            </td>
            <td />
            <td>
              NaN-NaN-NaN
            </td>
            <td>
              40
            </td>
          </tr>
        </tbody>
      </table>
      <div
        class="order-item-container"
      >
        <table
          class="order-item-table"
        >
          <thead>
            <tr
              class="order-item"
            >
              <th>
                Item Description
              </th>
              <th>
                Unit Price
              </th>
              <th>
                Quantity
              </th>
              <th>
                SubTotal
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              class="order-category"
            >
              <td>
                Fruits & Vegatables
              </td>
              <td>
                 
              </td>
              <td>
                 
              </td>
              <td>
                 
              </td>
            </tr>
            <tr
              class="item"
            >
              <td>
                Banana-Robusta
              </td>
              <td>
                40
              </td>
              <td>
                0
              </td>
              <td
                data-testid="item-amount"
              >
                0
              </td>
            </tr>
            <tr
              class="order-item-total"
            >
              <td>
                 
              </td>
              <td>
                 
              </td>
              <td>
                Total:
              </td>
              <td>
                40
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
